great_expectations/core/evaluation_parameters.py

```diff
@@ -219,11 +219,10 @@ def build_evaluation_parameters(
 
             # First, check to see whether an argument was supplied at runtime
             # If it was, use that one, but remove it from the stored config
-            if "$PARAMETER." + value["$PARAMETER"] in value:
-                evaluation_args[key] = evaluation_args[key][
-                    "$PARAMETER." + value["$PARAMETER"]
-                ]
-                del expectation_args[key]["$PARAMETER." + value["$PARAMETER"]]
+            param_key = f"$PARAMETER.{value['$PARAMETER']}"
+            if param_key in value:
+                evaluation_args[key] = evaluation_args[key][param_key]
+                del expectation_args[key][param_key]
```

Comment:
Do we want to delete `expectation_args[key]` here, or `evaluation_args[key]`? I see we make a deepcopy on L209, so I just want to check that we're not deleting something that might exist in a broader scope.